### ISSUE TITLE
Completing a task does not add a Done date

### ISSUE THREAD

**Issue by claremacrae** [sytone/obsidian-tasks-x#10]
I was excited to try out the new Tasks Status Types. Thank you for working on these.

## Expected Behavior

Installing this plugin and accepting its default settings, I would expect that clicking in the Task's checkbox would mark the task as done and add the Done date.

## Current Behavior

Instead of change `[ ]` to `[x]` it changes to `[/]`, which the settings say is 'In Progress'. And it therefore does not add the Done date, regardless of whether the user enabled that field or not.

It **looks** like it is a completed task, but it actually is not.

Problems with this behaviour include:

- It seems to silently impose on the user the requirement to install special styling to show the in-progress tasks - possibly even imposing Minimal or ITS Themes?
- It's a change in behaviour when the plugin says it's intended to be backwards-compatible with Tasks.

## Steps to Reproduce

- Install v2.4.2 of this plugin
- Create a task e.g. `- [ ] Do stuff`
- Switch to Reading mode
- Click on the task's check box

Result is: `- [/] Do Stuff`

## Context (Environment)

* Obsidian version: 0.15.2
* Tasks-X version: 2.4.2
* [x] I have tried it with all other plugins disabled and the error still occurs

## Possible Solution

- Change the default states to be as per the current Obsidian and Tasks plugin defaults, namely `[ ]` and `[x]` only
- And make it easy to add the 'In progress' state in the settings, for those who would like this feature

**Comment by sytone** [comment]
Thanks for the feedback, I will move the default progress from [ ] -> [x] -> [ ] and then add the custom state of in progress as an option to the UI. 

As for styling I am happy to support any style, I am using existing ones as I do not want to re-make the css styles and then keep them in sync with themes. Happy for any suggestions or ideas on this point. The default obsidian theme just has done and not done.

**Comment by claremacrae** [comment]
> Thanks for the feedback, I will move the default progress from [ ] -> [x] -> [ ] and then add the custom state of in progress as an option to the UI.

Perfect. Thanks very much.

> As for styling I am happy to support any style, I am using existing ones as I do not want to re-make the css styles and then keep them in sync with themes. Happy for any suggestions or ideas on this point. The default obsidian theme just has done and not done.

I'll have a think about options for that.

For example, if it's possible to adding custom task styling via snippets, we can point people to those in the settings, when they turn on the new states.

**Comment by sytone** [comment]
Yes, this is a option, if you know of a good snippet to use as a reference please shout out.

**Comment by claremacrae** [comment]
I looked briefly the other day without success. Will try to try again.

**Comment by sytone** [comment]
Found these two:

https://github.com/deathau/obsidian-snippets/blob/main/checkbox.css
https://github.com/RyzenFromFire/ryzen-checkboxes

**Comment by claremacrae** [comment]
This is a popular checkboxes set that I have started experimenting with:
https://github.com/SlRvb/Obsidian--ITS-Theme/blob/main/Guide/Alternate-Checkboxes.md
